packages/m/mkl/xmake.lua

```diff
@@ -109,42 +113,53 @@ package("mkl")
         package:add("links", package:is_arch("x64", "x86_64") and "mkl_blas95_" .. suffix or "mkl_blas95")
         package:add("links", package:is_arch("x64", "x86_64") and "mkl_lapack95_" .. suffix or "mkl_lapack95")
 
-        if package:has_tool("cc", "gcc", "gxx") then
-            local flags = {"-Wl,--start-group"}
-            table.insert(flags, package:is_arch("x64", "x86_64") and "-lmkl_intel_" .. suffix or "-lmkl_intel")
-            local threading = package:config("threading")
-            if threading == "tbb" then
-                table.insert(flags, "-lmkl_tbb_thread")
-                package:add("deps", "tbb")
-            elseif threading == "seq" then
-                table.insert(flags, "-lmkl_sequential")
-            elseif threading == "openmp" then
-                table.insert(flags, "-lmkl_intel_thread")
-                table.insert(flags, "-lomp")
-            elseif threading == "gomp" then
-                table.insert(flags, "-lmkl_gnu_thread")
-                table.insert(flags, "-lgomp")
-            end
-            table.insert(flags, "-lmkl_core")
-            table.insert(flags, "-Wl,--end-group")
-            package:add("ldflags", table.concat(flags, " "))
-        else
-            package:add("links", package:is_arch("x64", "x86_64") and "mkl_intel_" .. suffix or "mkl_intel_c")
-            local threading = package:config("threading")
-            if threading == "tbb" then
-                package:add("links", "mkl_tbb_thread")
-                package:add("deps", "tbb")
-            elseif threading == "seq" then
-                package:add("links", "mkl_sequential")
-            elseif threading == "openmp" then
-                package:add("links", "mkl_intel_thread", "omp")
-            elseif threading == "gomp" then
-                package:add("links", "mkl_gnu_thread", "gomp")
+        for _, toolkind in ipairs({"ld", "fcld"}) do
+            if package:has_tool(toolkind, "gcc", "gxx") or package:has_tool(toolkind, "gfortran") then
+
+                local var_ldflags = "ldflags"
+                local var_shflags = "shflags"
+                if package:has_tool(toolkind, "gfortran") then
+                    var_ldflags = "ldflags"
+                    var_shflags = "shflags"
```

Comment:
fcldflags, fcshflags?

Reply:
The problem is that the package fails with `toolchain=gfortran` because the linker flags are not  properly set. This gives an error for both OpenMP (#5255) and MKL packages.  If a special setting is necessary to get this to work with GFortran, could you send it to me, please?
We use `xmake --toolchain=gfortran`. A minimum example is provided in the bug report #5255 .
The most recent tests were done with *xmake v2.9.5+dev.9563841*

Reply:
I'll take a look at this issue, but need to wait a few days as I've been busy in these days.

Reply:
No problem, I am currently including the other changes you asked and will make the final changes as necessary, thank you very much!

Regarding the failure with Mac Intel x86_64, is there a workaround to make it work?  Should we explicitly exclude clang with a condition like:
```if not package:has_tool(toolkind, "clang", "clangxx") and package:has_tool(toolkind, "gcc", "gxx", "gfortran") then

Reply:
I have supported fcldflags, you can try https://github.com/xmake-io/xmake-repo/issues/5255#issuecomment-2365166806

Reply:
> Regarding the failure with Mac Intel x86_64, is there a workaround to make it work

if you use fcldflags instead of ldflags, it will also be fixed. so we need use fcldflags.

Reply:
I have merged that patch, you can use fcldflags and try it again.

Reply:
Thank you.  I was trying with the `ldflags` branch and have now switched to the new `dev`, but to no avail.
There is a problem with GFortran and the MKL package with the `gomp` option. It seems that using either ldflags or fcldflags now fail.

The new flags are:

`var_ldflags = fcldflags / flags = -Wl,--start-group -lmkl_intel_lp64 -lmkl_gnu_thread -lgomp -lmkl_core -Wl,--end-group`

Reply:
? what's problem for these flags?

Reply:
It fails during the configuration:

![image](https://github.com/user-attachments/assets/76b76b3c-a8d6-45f3-95e3-7c78de8f3621)

The verbose file does not provide much further insight.

Reply:
I have pushed the latest changes to the package with the edits you asked.  This version works fine for gcc/clang toolchains but fails with gfortran (Linux Debian testing, latest version of xmake from the dev branch).

Reply:
but it works on ci.

Reply:
Is there any fortran/gfortran test in the test suite?  From what I could see, only clang and gcc are tested, so the logic for gfortran is never controlled.

Reply:
The fortran code test is not currently supported. Can you provide a test example project, I will try it.